src/rotten-tomatoes/index.js

```diff
@@ -1,3 +1,100 @@
-function rottenTomatoes(grid) {}
+function rottenTomatoes(grid) {
+  const rowSize = grid.length;
+  const columnSize = grid[0].length;
+
+  // A function of getting the grid item
+  function getGridItem(row, column) {
+    return grid[parseInt(row)][parseInt(column)];
+  }
+
+  // Initializing the time
+  let time = 0;
+
+  // Initializing the starting nodes queue
+  let startingNodes = [];
+
+  // Initializing the visited nodes queue
+  let visitedNodes = [];
+
+  // Getting all the starting nodes
+  for (let row = 0; row < rowSize; row++) {
+    for (let column = 0; column < columnSize; column++) {
+      if (getGridItem(row, column) === 2) {
+        startingNodes.push([row, column]);
+      }
+    }
+  }
+
+  // Function for checking for the occurence of an array in another array
+  function checkForArray(parentArray, childArray) {
+    const child = JSON.stringify(childArray);
+
+    let contains = parentArray.some((point) => JSON.stringify(point) === child);
```

Comment:
This can make your program awfully slow. Avoid serializing stuff unnecessarily in your program. This code can work entirely without that.

Reply:
I had added it during some debugging earlier, but I just noticed its totally unnecessary in this case. First time I'm hearing of serialization. You don't miss anything! 😄 

Thanks for teaching me this one